Replicator/IncomingRev.cc

```diff
@@ -479,8 +482,11 @@ namespace litecore::repl {
     // Puller we are done.
     void IncomingRev::afterEvent() {
         Worker::afterEvent();
-        if ( _shouldNotifyPuller ) { _puller->revWasHandled(this); }
-        _shouldNotifyPuller = false;
+        if ( _shouldNotifyPuller.exchange(false) ) {
```

Comment:
Do you know why it needs be so complicated. Isn't _insertWasEnqueued all is needed? Can it be reset when reset() is called? We need to read it only in IncomingRev::finish is called. And afterEvent is called only if _insertWasEnqueued is true. It doesn't even need to be atomic because it's assigned value before it is enqueued, and is read in IncomingRev's queue.